src/main/resources/templates/db_scripts/migration/extract_credits_from_expenditures.sql

```diff
@@ -0,0 +1,59 @@
+-- Extract credits from expenditures in each budget
+-- 1. Calculate all credits values within budget fiscal year
+-- 2. Set to budget credits fields
+-- 3. Subtract from expenditures amount
+
+WITH credit_sums AS (
+    SELECT
+        budget.id AS budget_id,
+        SUM((trx.jsonb->>'amount')::numeric) AS total_credits
+    FROM ${myuniversity}_${mymodule}.budget AS budget
+    JOIN ${myuniversity}_${mymodule}.transaction AS trx
+        ON trx.jsonb->>'fiscalYearId' = budget.jsonb->>'fiscalYearId' AND
```

Comment:
can optimize this query.
These fields have separate columns and are foreign keys and RMB automatically creates indexes for foreign keys
trx.fiscalYearId
budget.fiscalYearId
trx.fromFundId
budget.fundId

Reply:
Updated,